corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js

```diff
@@ -106,18 +106,18 @@ FormplayerFrontend.module("Menus", function (Menus, FormplayerFrontend, Backbone
                 },
             });
 
-            if (model.isPersistentDetail) {
-                $('#select-case').hide();
-            } else {
-                $('#select-case').show();
-            }
-
             $('#select-case').off('click').click(function () {
                 FormplayerFrontend.trigger("menu:select", caseId);
             });
             $('#case-detail-modal').find('.js-detail-tabs').html(tabListView.render().el);
             $('#case-detail-modal').find('.js-detail-content').html(menuListView.render().el);
             $('#case-detail-modal').modal('show');
+
+            if (model.isPersistentDetail) {
+                $('#case-detail-modal').find('#select-case').hide();
```

Comment:
why can't you just use the `#select-case` id? usually if it's an id it should be the only instance on the page